Madagascar custom shapefile

nitinmagima opened the pull request:
Hi Aaron,

I update the yaml file to take in the new shapefiles where the new admin 0 and 1 shapefiles have been created using the admin 2 files. On running the dashboard in the server however I get this error. It doesn't seem like a code error but a server error?

Best,
Nitin 


<img width="827" height="509" alt="image" src="https://github.com/user-attachments/assets/fbe15850-da9d-4500-a089-6c970d8487a8" />

nitinmagima committed:
Update Madagascar configuration to use shapefile-based shapes

nitinmagima committed:
Update fbfmaproom-sample.yaml

aaron-kaplan committed:
Accept shapes in zip or unpacked form

aaron-kaplan committed:
Don't suppress error message when selected region isn't found

aaron-kaplan commented:
I found three distinct problems:
1. The zip file doesn't have the structure that fiona expects. I'm not sure how it should be structured, so for now I've unpacked the zip.
2. The Madagascar shapes are in a projected coordinate system, with units of meters, whereas the application expects a geographic coordinate system, with units of degrees. The Djibouti shapes worked because they use a geographic coordiate system.
3. A bug in the application code caused the error mesage to be silently discarded instead of being printed on the console.

I've pushed changes to this branch to fix 1 and 3. @nitinmagima please pull my changes, and then see if you can figure out how to reproject/deproject the shapes.

Also please share your script for constructing and deprojecting the shapes, because eventually I'd like to put that logic in the application. If the application can handle shapes in whatever form we find them, rather than requiring pre-processing, it will make it easier to add new shapes.

mm5330 commented:
> @mm5330 @nitinmagima what is going on? Github says Max just pushed two commits authored by Nitin that duplicate work Max did on a different branch.
> 
> There should be nothing about Ethiopia on this branch. This branch is for Madagascar changes.

Oh weird, this wasn't me. Not sure how that happened...

nitinmagima commented:
Hi Aaron,

So changing the coordinate system worked but the boundaries themselves look
weird. I'm wondering if that has to do with the data. Both the zip files
and the shapefiles work now but there's an error.

Traceback (most recent call last):
  File "/home/magima/python-maprooms/fbfmaproom/fbfmaproom.py", line 1144,
in table_cb
    raise Exception("No region found")
Exception: No region found

[image: image.png]


Here's the code I'm using -
https://drive.google.com/file/d/1G4oUoik09moCe7QvvLnmaUGfeFKd3PuY/view?usp=sharing

I pulled in changes from the main branch, I'm guessing that's where Max's
comments got pulled in.

Best,
Nitin

On Thu, Oct 30, 2025 at 2:53 PM Aaron Kaplan ***@***.***>
wrote:

> aaron-kaplan left a comment (iridl/python-maprooms#575) I found three
> distinct problems: The zip file doesn't have the structure that fiona
> expects. I'm not sure how it should be structured, so for now I've unpacked
> the zip. The Madagascar shapes
> ZjQcmQRYFpfptBannerStart
> This Message Is From an External Sender
> This message came from outside your organization.
>
> ZjQcmQRYFpfptBannerEnd
> *aaron-kaplan* left a comment (iridl/python-maprooms#575)
> <https://urldefense.com/v3/__https://github.com/iridl/python-maprooms/pull/575*issuecomment-3469592438__;Iw!!BDUfV1Et5lrpZQ!Qz6_PFBBhyjU04WzH0EronD38rjTXqVOZjP-EesrPt30LuqxpBN5JzBKTVX5C2YWnGDV94o0OyPc-yddHAdHcT1onPzrRw$>
>
> I found three distinct problems:
>
>    1. The zip file doesn't have the structure that fiona expects. I'm not
>    sure how it should be structured, so for now I've unpacked the zip.
>    2. The Madagascar shapes are in a projected coordinate system, with
>    units of meters, whereas the application expects a geographic coordinate
>    system, with units of degrees. The Djibouti shapes worked because they use
>    a geographic coordiate system.
>    3. A bug in the application code caused the error mesage to be
>    silently discarded instead of being printed on the console.
>
> I've pushed changes to this branch to fix 1 and 3. @nitinmagima
> <https://urldefense.com/v3/__https://github.com/nitinmagima__;!!BDUfV1Et5lrpZQ!Qz6_PFBBhyjU04WzH0EronD38rjTXqVOZjP-EesrPt30LuqxpBN5JzBKTVX5C2YWnGDV94o0OyPc-yddHAdHcT2BpSj9qg$>
> please pull my changes, and then see if you can figure out how to
> reproject/deproject the shapes.
>
> Also please share your script for constructing and deprojecting the
> shapes, because eventually I'd like to put that logic in the application.
> If the application can handle shapes in whatever form we find them, rather
> than requiring pre-processing, it will make it easier to add new shapes.
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/iridl/python-maprooms/pull/575*issuecomment-3469592438__;Iw!!BDUfV1Et5lrpZQ!Qz6_PFBBhyjU04WzH0EronD38rjTXqVOZjP-EesrPt30LuqxpBN5JzBKTVX5C2YWnGDV94o0OyPc-yddHAdHcT1onPzrRw$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AFK3DJL4DK2LBHF226WKWIT32JNC3AVCNFSM6AAAAACKMBBWMKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTINRZGU4TENBTHA__;!!BDUfV1Et5lrpZQ!Qz6_PFBBhyjU04WzH0EronD38rjTXqVOZjP-EesrPt30LuqxpBN5JzBKTVX5C2YWnGDV94o0OyPc-yddHAdHcT21bjMjUQ$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Best,
________________________________
*Nitin Magima (he/him/his) *
Staff Associate II, Officer of Research
MPA DP | FSA Credential Holder | AWS Certified Cloud Practitioner
National Center for Disaster Preparedness (NCDP) <http://ncdp.columbia.edu>
| Affiliate of IRI
Columbia Climate School | Columbia University
Office:  212.853.NCDP
E-mail: ***@***.***
Medium ***@***.***> | Github
<https://github.com/nitinmagima> | LinkedIn
<https://www.linkedin.com/in/nitin-magima/>

aaron-kaplan committed:
Rename shapefile

aaron-kaplan committed:
Re-enable clipping for Madagascar

aaron-kaplan committed:
Nitin's Madagascar shapes script

aaron-kaplan committed:
Remove holes caused by neighboring regions whose edges don't exactly match up

aaron-kaplan committed:
Change file names

aaron-kaplan committed:
Remove unnecessary ALL layer

aaron-kaplan committed:
Freeze conda environment for mada shapes script

aaron-kaplan commented:
I've added Nitin's script to this repo, and modified it to remove the holes that `dissolve` left in the shapes. It seems to be working now:

http://mako.iri.columbia.edu:9998/fbfmaproom/madagascar

A couple things to note:
1. The shapes are quite slow to load even for me, located in the US. It remains to be seen how usable this is for people in Madagascar.
2. We changed the outer boundary to include the entire country (used to only include the four southernmost regions), but we haven't expanded the obs data to cover the entire country, so you get errors if you click too far north. @dantesab9 if you want to fix this, you can do so by modifying the Y (latitude) bounds in the Ingrid URLs (which now live in the config file) and rerunning `fbf-update-data.py`. Alternatively, since we're already modifying the shapefile, it would be easy enough to change the script to exclude regions to the north that aren't of interest. That would probably also help with the speed.

@dantesab9 let me know what you want to do.

dantesab9 commented:
Hi Aaron, thank for this update too, the map look very good! About the obs data coverage, I think it is best to extend to the full country since we will use some districts that are not in the prior "national" polygon and also we may need to provide data for more northern districts that are part of the other, newer, red cross project. I'll take a look at the latitude question and also start working in updating the polygon IDs in the markdowns.

nitinmagima committed:
updated the observation links to include the whole country

nitinmagima committed:
Fix Madagascar ENACTS dataset URLs from http to https

nitinmagima commented:
Hi @aaron-kaplan,

After discussing with @dantesab9, please merge this branch. I've updated the links for the observations datasets as well and updated the datasets in the server. Please sync the data as well.

Best,
Nitin

aaron-kaplan commented on fbfmaproom/fbfmaproom.py:
```diff
@@ -344,9 +344,13 @@ def retrieve_shapes(
+        else:
+            prefix = ""
+        file_arg = f"{prefix}{CONFIG['data_root']}/{sc['file']}"
+        with fiona.open(file_arg, layer=sc["layer"]) as collection:
```
Do we still need this change? It looks like you used zip files in the end.

nitinmagima replied:
Actually you're right, since we're only using zip files, we don't right? Should I remove this from the commit?

aaron-kaplan replied:
Yes please. Not that it's a bad idea, but since it would affect all the countries, it would need more extensive testing and a new build of the container, which I'd rather avoid if it's not necessary.

nitinmagima replied:
Makes sense! I've removed that block.

nitinmagima committed:
Updated fbfmaproom.py

nitinmagima committed:
Updated fbfmaproom.py

aaron-kaplan commented on fbfmaproom/fbfmaproom-sample.yaml:
```diff
@@ -137,25 +137,24 @@ countries:
         zoom: 7
         rotation: 0
         marker: [45.5, -24]
-        clip: no
```
What was the reason for this change? (Does anyone remember?)

aaron-kaplan commented:
I've deployed this. As I noted in a previous comment back in November, it's extremely slow to load, to the point that I thought it was broken. I started looking in the log file for errors and then the map finally appeared.

nitinmagima commented:
I noticed that as well. I'm wondering if it has to do with the shapefile itself since we don't face this problem with other countries where we use the humdata shapefiles.
